### ISSUE TITLE
Lua Error created when right clicking or moving the scroll wheel

### ISSUE THREAD

**Issue by Timtee** [wiremod/wire#2345]
I don't really know what part of Wiremod exactly this issue is tied to, but I know that whatever it's tied to, the error in the console and in the problems menu in gmod says the same thing:

[Wiremod] lua/weapons/gmod_tool/stools/wire_adv.lua:726: attempt to call field 'GetActiveTOOL' (a nil value)
    1. fn - lua/weapons/gmod_tool/stools/wire_adv.lua:726
        2. unknown - lua/ulib/shared/hook.lua:109

And this particular error ONLY appears when you scroll thru your weapons, or try to use the alt function on them. (aka, right click.)

Also there is a second set of errors that only appears whenever you start up a map.
[Wiremod] lua/weapons/gmod_tool/stools/wire_adv.lua:192: attempt to call field 'SetupSingleplayerClickHacks' (a nil value)
    1. unknown - lua/weapons/gmod_tool/stools/wire_adv.lua:192
        2. include - [C]:-1
            3. unknown - gamemodes/sandbox/entities/weapons/gmod_tool/stool.lua:142
                4. include - [C]:-1
                    5. unknown - gamemodes/sandbox/entities/weapons/gmod_tool/shared.lua:353
                        6. include - [C]:-1
                            7. unknown - gamemodes/sandbox/entities/weapons/gmod_tool/init.lua:10

[Wiremod] lua/weapons/gmod_tool/stools/wire_debugger.lua:238: attempt to call field 'SetupSingleplayerClickHacks' (a nil value)
    1. unknown - lua/weapons/gmod_tool/stools/wire_debugger.lua:238
        2. include - [C]:-1
            3. unknown - gamemodes/sandbox/entities/weapons/gmod_tool/stool.lua:142
                4. include - [C]:-1
                    5. unknown - gamemodes/sandbox/entities/weapons/gmod_tool/shared.lua:353
                        6. include - [C]:-1
                            7. unknown - gamemodes/sandbox/entities/weapons/gmod_tool/init.lua:10

Also all three of these errors appear regardless of whether your in singleplayer or multiplayer. And I'm not entirely sure whether this is fixable, or simply just a semi~corrupted install of Wiremod from the workshop that I have for whatever reason.

**Comment by Divran** [comment]
If you open this file <https://github.com/wiremod/wire/blob/master/lua/wire/tool_loader.lua#L356>
and scroll down a bit you will see this is where all these functions that your error messages are complaining about are defined.

Then, if you go to the top of the wire_adv file here <https://github.com/wiremod/wire/blob/master/lua/weapons/gmod_tool/stools/wire_adv.lua#L6> 
this is where the code in other file I linked is executed

The only reason I could think of that the functions defined in that file are not accessible immediately after would be if there is another error message that's preventing the `tool_loader` file from executing properly.

**Comment by Timtee** [comment]
Hmmm... Probably, Unfortunately tho, Figuring out what could be causing it to do such would be no easy task for me... (i have like 700 addons.) But seeing as it is solely contained to the base mod... it is curious that Wiremod is trying to look for that ends up generating the errors because it can't find that it's looking for... let alone why. Or if it's trying to find something within the menus from one of the addons I have for wiremod, that isn't found and could be causing the problem then... Unsure otherwise.

I'll also say that toggling the flashlight on and off also causes the error to pop up too for the first of the 3 mentioned (aka the one that appears every time you right click, toggle flashlight, or scroll thru the weapons you have on you.) But... otherwise unsure why it occurs only on these exact things, and not inside of the spawn menu. (However, the other thing, aka the stuff accessible via C by default, also allows the error to appear when you right click on stuff because you can still use the weapons there of course.)

**Comment by Divran** [comment]
All you can do is to uninstall all addons and then reinstall them one by one while relaunching gmod (or at least closing and reopening single player / local server) in between each install and then check when it breaks. Then you'll know which addon does it.

Unfortunately, 700 is about 690 too many, so you probably don't want to do that. Your other best bet is to just uninstall all of them except wiremod and just not reinstall them cause who needs anything other than wiremod anyway

**Comment by Timtee** [comment]
Most of the addons I have for various reasons. Tho a fair bit of them are maps or player models. Obvious stuff allot of people would have a bunch of. But... I'll at least see if reinstalling Wiremod does anything (of which I actually thought of on my own before I even bothered to tab out of the game just a moment ago whilst sitting in the main menu for a bit doing stuff.) But uh... I personally don't really use Wiremod too much beyond just messing around the things other people have made because honestly, I just think it's cool with what is possible using Wiremod alone.

**Comment by thegrb93** [comment]
> Or am I the only instance thusfar of this issue currently?

Yeah you are

**Comment by Timtee** [comment]
> > Or am I the only instance thusfar of this issue currently?
> 
> Yeah you are

Fair enough, and honestly makes sense that I'd probably be the only one currently. As of right now, deciphering what is going on is honestly troubling because well... I'm not getting anything from the console because it's often too full with junk from other addons that whatever causes it is so early in loading, or early enough to get completely hidden by more recent stuff in the console...

**Comment by Timtee** [comment]
hold up. I just noticed something upon just launching GMod once more today (and right now at that.)
Wiremod Devices(Pre-release) also has the tool loader in it... My guess is that it, and Wiremod itself are both fighting one another with the tool loader.
And I bring this info simply because I noticed that in the problems portion of the problems menu, that one of the things is showing that they have conflicting files. Which... would most likely be the source of the lua errors. Surprised tho that both aren't making the error instead of just Wiremod itself.
